LayoutTests/fast/css/lang-matching-document-invalidation.html

```diff
@@ -0,0 +1,231 @@
+<!DOCTYPE html>
+<script src="../../resources/js-test.js"></script>
+<body>
+<script>
+var container, child, otherFrame;
+var testNumber = 0;
+
+description("This tests invalidation of :lang selectors when the document element's lang attribute changes and a descendant element is using the 'explicit language matching document element language' optimization");
+jsTestIsAsync = true;
+
+function matchedLang(element) {
+    for (let lang of ["en", "de", "zh"]) {
+        if (element.matches(`:lang(${lang})`))
+            return lang;
+    }
+    return "";
+}
+
+function shouldMatchLang(elementName, lang) {
+    shouldBe(`matchedLang(${elementName})`, `"${lang}"`);
+}
+
+function test(f) {
+    document.documentElement.lang = "en";
+    try {
+        f();
+    } catch (ex) {
+        console.log(ex);
+        finishJSTest();
+    }
+    container = null;
+    child = null;
+    debug("----");
+}
+
+async function testWithOtherDocument(f) {
+    document.documentElement.lang = "en";
+    otherFrame = document.createElement("iframe");
+    await new Promise(function(resolve) {
+        otherFrame.onload = () => resolve();
+        otherFrame.srcdoc = `<!DOCTYPE html><html lang=en><body>`;
+        document.body.append(otherFrame);
```

Comment:
No need to do this. Just insert iframe & start accessing contentDocument. Everything happens synchronously in WebKit/Blink.